### DISCUSSION TITLE
How can I change the size of figures?

### DISCUSSION THREAD

**Discussion by Junoh-Kang** [alshedivat/al-folio#1875] [Q&A] [answered]
How can I change the size of figures?

**Comment by george-gca** [comment]
By changing its width/height with css. Where exactly are you trying to do this? Depending on the section it might have better solutions than others.

**Comment by Junoh-Kang** [comment]
I tried to embed figures in posts (blog). 

I solved the problems somehow by following:
```
|![]({{ "path" | relative_url }}){:style="margin:auto; display:block;width:60%; height:auto;"}|
|:--|
| captions |
```

However, I wonder if there are better solutions like using figure.html, like the example posts in al-folio.
```
<div class="row mt-3">
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/9.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
    <div class="col-sm mt-3 mt-md-0">
        {% include figure.html path="assets/img/7.jpg" class="img-fluid rounded z-depth-1" %}
    </div>
</div>
<div class="caption">
    A simple, elegant caption looks good between image rows, after each row, or doesn't have to be there at all.
</div>
```

**Reply by george-gca** [reply]
The recommended way is to use `figure.html`, since it handles generating and [displaying](https://github.com/alshedivat/al-folio/blob/85f2590a2119ec57063845d628221155044ebea8/_includes/figure.html#L7) appropriate versions of the images.

**Comment by DMatsliyah** [comment]
I have zero experience with HTML and CSS (which is why I couldn't apply @george-gca 's suggestion. The easiest way I found was to just specify column numbers. I can't remember exactly where I read it but it seems like the total number of columns is 12 (the 100% essentially). so specifying any other lower number of columns should reduce the size of the image.

In one case where the image shown was larger than what I wanted I simply reduced to use `col-sm-7` instead of `col-sm`.

I hope this helps.

I just re-read your post and realized you're talking about 'post' type pages, and I was looking at 'page' type pages. I don't know if there's any difference that would make my solution specific to this layout.

**Reply by george-gca** [reply, marked as answer]
You can basically mix the solutions. The `figure.html` solution allows you to set the css class of the figure, as you can see in the example:
```html
{% include figure.html path="assets/img/9.jpg" class="img-fluid rounded z-depth-1" %}
```
So, you could also set `col-sm-7` for the figure here. As for this grid system, it is a bootstrap thing. You can [read more about it in the docs](https://getbootstrap.com/docs/4.6/layout/grid/).
